### PR TITLE
Update the Technical Infrastructure section

### PR DESCRIPTION
**Pull request by matthew-morris-cessda**
This PR focuses on improving wording and updating terminoligy so that it reflects current usage. The template Jenkinsfiles have been updated as well.

See #105.

### CODE REVIEW

File: technical-infrastructure/deploying-core-products.md [hunk 1]
```diff
@@ -19,41 +19,48 @@ This is the procedure for deploying a new version of a CESSDA core {% include gl
...
-Creates an issue in the tool-specific issue tracker in GitHub,
-changes the status to OPEN and assigns it to the Content Contact for the tool.
+Once all pre-release activities have been concluded and the release has been approved, the release can be performed.
 
-- Recommend pattern for issue title is *'Release version X.Y.Z of \<tool name\> to production'*
+- The source code repositories must [tagged](https://git-scm.com/book/en/v2/Git-Basics-Tagging) with the release version number. The format used is X.Y.Z.
+- The service maintainer verifies that the tagged build is successfully deployed to the staging environment.
 
-- For traceability, the version number of the release **must** be included in the issue.
+Deploying to production happens via production deployment jobs, which are postfixed with `deploy.prod` in Jenkins. These jobs automatically tag Docker images as production images, then calls the deployment script of the product with the tagged images. The deployment script will deploy the tagged images to the production cluster. This is done by the platform maintainer.
```
Comment (john-shepherdson) [inline technical-infrastructure/deploying-core-products.md:50]: calls -> call

File: technical-infrastructure/deploying-core-products.md [hunk 2]
```diff
@@ -19,41 +19,48 @@ This is the procedure for deploying a new version of a CESSDA core {% include gl
 
 See the 'Product Coordination' document for definition of roles.
 
-- Technical Contact
+- Service Maintainer
```
Comment (john-shepherdson) [inline technical-infrastructure/deploying-core-products.md:22]: Service Maintainer -> Technical Maintainer

File: technical-infrastructure/deploying-core-products.md [hunk 3]
```diff
@@ -19,41 +19,48 @@ This is the procedure for deploying a new version of a CESSDA core {% include gl
...
+
+- The recommend pattern for issue title is *'Release version X.Y.Z of \<tool name\>'* where X.Y.Z is the release version number
+- The issue should state the general purpose of the release, as well as the most significant issues that either need to be, or have been, resolved
+- The issue should also contain links to issues regarding the following subtasks
+  - Creating the release notes, which contain a list of all issues and PRs that are part of the release
+  - Updating user facing documentation with new features/changed behaviour
+- The issue should include a checklist of steps that are necessary for a release
+  - Backups before release
+  - Upgrade procedures for dependency updates
+- The issue should also contain links to the release checklist and the release report
```
Comment (john-shepherdson) [inline technical-infrastructure/deploying-core-products.md:40]: Reserve a DOI in Zenodo and add it to the ChangeLog

File: technical-infrastructure/gcp-main-project-structure.md [hunk 4]
```diff
@@ -14,28 +14,24 @@ See [Naming Conventions]({% link technical-infrastructure/naming-conventions.md
...
-monitoring and management services. They are separated within the cluster by various namespaces
-(`cdc`, `cvs`, `dvs`, `eqb`, `mgmt`, `monitoring`).
+The development cluster is where newly built components are deployed. Components are deployed directly after a successful build. Integration tests are run against components deployed to this cluster. Applications may break at any time in this cluster.
 
 ![GCP Main Project Structure Development](../images/gcp-main-project-structure-development.png)
 
-The `staging-cluster` contains the components of the  CDC, CVS, DVS and EQB tools plus some
-monitoring and management services. They are separated within the cluster by various namespaces
-(`cdc`, `cvs`, `dvs`, `eqb`, `mgmt`, `monitoring`).
+The staging cluster is deployed to after integration and is used as a stable basis for user testing. Components are deployed to this cluster after integration tests have passed, which ensures this stability. Stability of applications in the staging cluster is maintained on a best effort basis.
```
Comment (john-shepherdson) [inline technical-infrastructure/gcp-main-project-structure.md:27]: Either stability is 'ensured' or is 'best efforts' but not both.

File: technical-infrastructure/repository-types.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+# {{ page.title }}
+
+## Terminology
+
+See [Naming Conventions]({% link technical-infrastructure/naming-conventions.md %}).
+
+## Overview
+
+There are three types repositories used in the CI/CD process; {% include glossary.html entry="(component)" text="component" %} repositories and Deployment repositories.
```
Comment (john-shepherdson) [inline technical-infrastructure/repository-types.md:15]: Deployment -> deployment

Also, add glossary entry for 'deployment'

File: technical-infrastructure/technical-infrastructure-overview.md [hunk 6]
```diff
@@ -31,21 +31,13 @@ The structure of each GCP project follows the same set of principles:
...
+The Google Kubernetes Engine (GKE) is an environment for deploying containerized (i.e. Dockerized) applications.
 A Docker container provides an isolated context in which an application can run with its own environment.
-At every deployment instance, container images are built and pushed to the Google Cloud registry associated
-with the GCP Production project.
-Docker containers use these images when running the applications.
-This enables GKE to provide rapid application development by making it easy to deploy, update,
-and manage CESSDA's products.
-GKE can operate both in stateless mode, and also can be configured to deploy backing components like persistent volumes/storage,
-and even run a database in the cluster.
+Docker containers run Docker images, which is a snapshot of an application and all its dependencies.
```
Comment (john-shepherdson) [inline technical-infrastructure/technical-infrastructure-overview.md:39]: Docker containers run Docker images -> A Docker container runs a Docker image

File: technical-infrastructure/technical-infrastructure-overview.md [hunk 7]
```diff
@@ -59,71 +51,52 @@ See [GCP Main Project Structure]({% link technical-infrastructure/gcp-main-proje
...
-To put this in the context of a specific tool, the CDC development instance is deployed to the `development-cluster`
-of the CESSDA Development GCP Project,
+For example, the development instance of CDC is deployed to the `development-cluster` of the CESSDA Development GCP Project,
 the staging instance of CDC is deployed to the `staging-cluster` of the CESSDA Development GCP Project,
-and the CDC production instance is deployed to the `production-cluster` of the CESSDA Production GCP Project.
-The same goes for the development, staging and production instances of CVS, DVS and EQB.
-Therefore the `development-cluster` acts as the development environment, the `staging-cluster`
-acts as the staging environment and the `production-cluster` acts as the production environment.
+and the production instance of CDC is deployed to the `production-cluster` of the CESSDA Production GCP Project.
+The same goes for the development, staging and production instances of other products such as CVS, DVS and EQB.
```
Comment (john-shepherdson) [inline technical-infrastructure/technical-infrastructure-overview.md:85]: such as CVS, DVS and EQB -> such as CVS and CMV

File: technical-infrastructure/technical-infrastructure-overview.md [hunk 8]
```diff
@@ -132,11 +105,12 @@ See [Jenkinsfile template for JDK11 Maven Projects]({% link technical-infrastruc
...
-When commits are made to master, a successful build deploys to the development cluster.
-After further testing, it is automatically deployed to the staging cluster.
-If ready to go live, it is manually deployed to the `production-cluster`
-([datacatalogue.cessda.eu](https://datacatalogue.cessda.eu/)).
+Each component of CDC is built from the main branch of the Git repository.
+
+- Jenkins automatically builds the main branch when commits are made to it.
+- A successful build is deployed to the development cluster.
+- Integration tests are run against the deployed components of the development cluster. If these are successful the components are deployed together on the staging cluster.
+- If ready to go live, the components are deployed manually to the `production-cluster`.
```
Comment (john-shepherdson) [inline technical-infrastructure/technical-infrastructure-overview.md:113]: If ready to go live -> When ready to go live

File: technical-infrastructure/template-jenkinsfile-for-jdk11-maven-projects.md [hunk 9]
```diff
@@ -12,37 +12,48 @@ See [Naming Conventions]({% link technical-infrastructure/naming-conventions.md
 
 ## Overview
 
-This shows the general form of a Jenkinsfile used with a Maven project based on JDK11.
+This shows an example of a Jenkinsfile that could be used with a Maven project based on JDK11.
```
Comment (john-shepherdson) [inline technical-infrastructure/template-jenkinsfile-for-jdk11-maven-projects.md:15]: Is JDK11 still relevant?

**Review by john-shepherdson — COMMENTED** [review]
See inline comments

File: technical-infrastructure/template-jenkinsfile-for-jdk11-maven-projects.md (hunk 9 above)
Comment (matthew-morris-cessda) [reply technical-infrastructure/template-jenkinsfile-for-jdk11-maven-projects.md:15]: Yes, it is. CVS still uses JDK 11 and it's a standard baseline within the Java community.

**Review by john-shepherdson — APPROVED** [review]

**Pull request merged**